raiden/api/python.py

```diff
@@ -808,26 +843,40 @@ def get_token_network_events(
             if event.get('args'):
                 event['args'] = dict(event['args'])
 
-                # the channel_identifier is a hash
-                encode_byte_values(event['args'])
+        returned_events.sort(key=lambda evt: evt.get('block_number'), reverse=True)
+        return returned_events
 
-            hexbytes_to_str(event)
+    def get_token_network_events_raiden(
+            self,
+            token_address,
+            from_block,
+            to_block='latest',
+    ):
+        """Returns a list of internal events coresponding to the token_address."""
+        if not is_binary_address(token_address):
+            raise InvalidAddress(
+                'Expected binary address format for token in get_token_network_events_blockchain',
+            )
 
+        returned_events = []
         raiden_events = self.raiden.wal.storage.get_events_by_block(
             from_block=from_block,
             to_block=to_block,
         )
 
-        # Here choose which raiden internal events we want to expose to the end user
         for block_number, event in raiden_events:
-            if isinstance(event, EVENTS_PAYMENT_HISTORY_RELATED):
-                new_event = {
-                    'block_number': block_number,
-                    'event': type(event).__name__,
-                }
-                new_event.update(event.__dict__)
-                returned_events.append(new_event)
-
+            new_event = {
+                'block_number': block_number,
+                'event': type(event).__name__,
+            }
+            if hasattr(event, 'transfer'):
+                if event.transfer.token == token_address:
+                    event.transfer = repr(event.transfer)
+            elif hasattr(event, 'token'):
```

Comment:
Above you implemented this a bit differently, would it make sense to extract these functions?

Reply:
[Here](https://github.com/raiden-network/raiden/blob/c22e066c3c9c0049fad552e7d06ca2f3a402100d/raiden/api/python.py#L773-L813) I had the requirement to filter by partner address, while [here](https://github.com/raiden-network/raiden/blob/c22e066c3c9c0049fad552e7d06ca2f3a402100d/raiden/api/python.py#L849-L881) there was no need for that. 

You are right about the fact that this part: 
https://github.com/raiden-network/raiden/blob/c22e066c3c9c0049fad552e7d06ca2f3a402100d/raiden/api/python.py#L872-L879
can be moved in a helper function, and used in both ``get_token_network_events_raiden`` and ``get_channel_events_raiden``. 

Should I do that?